### ISSUE TITLE
Bug when off effect used with canvas enabled on an effect with a different buffer size.

### ISSUE THREAD

**Issue by qualar** [xLightsSequencer/xLights#2599]
NOTE: IF YOU DO NOT ATTACH A SEQUENCE THAT DEMONSTRATES YOUR PROBLEM THEN THERE IS A HIGH PROBABILITY YOUR ISSUE WILL JUST BE CLOSED AND NOT INVESTIGATED.

**Describe the bug**
If you have an off effect with canvas enabled above an effect with a different buffer size or rotation you end up with lots of visual artefacts.

**To Reproduce**

1. Place an off effect on layer 1 and enable canvas mode and transparency
2. Place Butterfly effect on layer 2 and make the buffer half size or set the buffer to a different rotation.
3. You will now see lots of pixels flashing randomly. 

**Expected behaviour**
The effect should show correctly

**Screenshots**

**Versions (please complete the following information):**
 - OS: Windows 10 [Version 10.0.19042.867]
 - xLights version : 2021.12 (64bit)

**Attachments**
Here is a sequence showing the problem
https://1drv.ms/u/s!ArKnPav27dHwkel9gYMqvLTVdo0W8A?e=m5AEoN

**Comment by AzGilrock** [comment]
Sounds more like a feature than a bug.  You are using canvas mode and then pulling in undefined pixels.  So define them to what you want them to be.

**Comment by qualar** [comment]
:-)  I submitted at Keith's request when I showed him it in the Zoom Room yesterday.   But i suspected it was the undefined areas of the buffer because similar things happen if you use the off effect to freeze frame and don't specify the same buffer dimensions.

**Comment by AzGilrock** [comment]
Yeah well he fixes a lot of things I think the user should just learn to deal with....lol.  Mainly cause I don't like adding code that slows down the program to fix edge cases that the user could fix on their own.

**Comment by derwin12** [comment]
With the rewrites to ISPC and the OFF effect, this no longer replicates.
